src/choose-location/choose-location.component.tsx

```diff
@@ -48,160 +44,47 @@ export default function ChooseLocation(props: ChooseLocationProps) {
           }
         })
         .catch(createErrorHandler());
-    }
-    return () => abortController.abort();
-  }, [isSubmitting]);
+    },
+    [referrer]
+  );
+
   React.useEffect(() => {
     const ac = new AbortController();
-    changeLocationData({ locationResult: props.loginLocations });
-    if (props.loginLocations.length > 100) {
-      if (searchTerm) {
-        searchLocationsFhir(searchTerm).then((locs) => {
-          changeLocationData({ locationResult: locs.data.entry });
-          if (isEmpty(locs.data.entry)) {
-            changeLocationData({ emptyResult: true });
-          } else {
-            changeLocationData({ emptyResult: false });
-          }
-        }, createErrorHandler());
-      }
-    } else if (searchTerm) {
-      filterList(searchTerm);
-    } else {
-      changeLocationData({ emptyResult: false });
-      changeLocationData({ locationResult: props.loginLocations });
-    }
-    return () => ac.abort();
-  }, [searchTerm, props.loginLocations]);
+    const sub = getCurrentUser().subscribe((user) => {
+      setCurrentUser(user ? user.display : currentUser);
+    }, createErrorHandler());
 
-  const search = debounce((location) => {
-    setSearchTerm(location);
-  }, searchTimeout);
+    queryLocations("", ac).then(
+      (locations) => setLoginLocations(locations),
+      createErrorHandler()
```

Comment:
I take it this thing is not a normal promise? I've never seen a two-argument call to `then` before...

Reply:
Note this was just from the original code.

Independent, yes this is the normal way to catch the original promise or otherwise continue.

So

```ts
Promise.then(resolve, reject)
```

is different to

```ts
Promise.then(resolve).catch(reject)
```

Reason: The latter is also catching if the resolve throws. The former only if the original promise rejected.

Reply:
TIL! Thank you!